router/functionHandler.go

```diff
@@ -52,11 +64,12 @@ func (fh *functionHandler) getServiceForFunction() (*url.URL, error) {
 type RetryingRoundTripper struct {
 	maxRetries    int
 	initalTimeout time.Duration
+	transport     http.Transport
 }
 
 func (rrt RetryingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	timeout := rrt.initalTimeout
-	transport := http.DefaultTransport.(*http.Transport)
+	transport := &rrt.transport
```

Comment:
@soamvasani 
I can use a experiment to reproduce but I cannot fully explain it yet. I will dig into the reason later. Please use this [yaml](https://gist.github.com/yqf3139/e7a39dace9335ac28fb8d3552d291410). The code is [here](https://gist.github.com/yqf3139/e9a9ec3bf33452c55e07aa135eb56619).

I create one proxy pod and two server pod. When we visit the proxy pod, it visits the server service and proxy the result back. We should see different uuid of the server when curling the node_ip:31320/proxy.

If we use the `use-pointer` tag image, the response is the same among all the response. `use-copy` is the correct one, which uses a copy of transport instead of a pointer.
I tried to set the `keep-alive` to 0 in the default transport but it is not working.